apps/vscode/src/providers/format.ts

```diff
@@ -224,8 +233,17 @@ async function formatBlock(doc: TextDocument, block: TokenMath | TokenCodeBlock,
           new Position(edit.range.end.line + block.range.start.line + 1, edit.range.end.character)
         );
         return new TextEdit(range, edit.newText);
-      })
-      .filter(edit => blockRange.contains(edit.range));
+      });
+
+    // Bail if any edit is out of range. We used to filter these edits out but
+    // this could bork the cell.
+    if (edits.some(edit => !blockRange.contains(edit.range))) {
```

Comment:
Unrelated but I'm pretty sure applying edits partially can bork our cells 😬

Reply:
I pulled down this PR and followed the reproduction steps in https://github.com/quarto-dev/quarto/issues/745 (which this PR should address). I ran into a new error that prevented any formatting from occuring:

Screenshot 2025-09-04 at 12 12 30 PM

I reverted this code change locally (bailing if edits are out of range) and formatting works again and the problem described in #745 is addressed.